### DISCUSSION TITLE
Async loading of modal content

### DISCUSSION THREAD

**Discussion by ivan-plotnikov** [inertiaui/modal#24] [Ideas]
Now when a modal window is called, a request is made to the server and only then is the modal rendered. Depending on the speed of the internet connection, this can take from a few milliseconds to a few seconds. 

I think it would be useful to have a customisable mode where the modal window opens immediately after the call, and the content is loaded afterwards (a preloader or skeleton could be used for the loading time).

This would be especially useful for users from countries with slow internet.

**Comment by pascalbaljet** [comment]
Yeah, that would be interesting! For now, you can use the [loading state](https://inertiaui.com/inertia-modal/docs/basic-usage#loading-state) on the `ModalLink` component. I will convert this to a discussion to keep the issues clean.

**Reply by ivan-plotnikov** [reply]
Adding prefetch from inertia 2 may solve this problem 
https://v2.inertiajs.com/prefetching
